js-project-guidelines.md

```diff
@@ -64,55 +188,87 @@ lib
 
-// WIP
+#### ...for consumers
 
-## Building a dist version for the browser
+Consumers of our modules should not have to think about our set up,
+unless they want to.
+So we provide four different ways of consuming our code.
 
-// WIP
+For use in the browser through script tags there is regular and a minified version in the npm release.
+An example of using those is through [npmcdn](https://npmcdn.com/),
 
-- Browser building: [webpack](https://webpack.github.io/) or [browserify](http://browserify.org/).
+```html
+<script src="https://npmcdn.com/ipfs-api/dist/index.js"></script>
+<script src="https://npmcdn.com/ipfs-api/dist/index.min.js"></script>
+```
 
-## Continuous integration
+If you install the module through npm and require it, you receive the
+ES5 version ready to be used in Node.js or a module bundler like browserify.
 
-Our picks are:
+```js
+var API = require(‘ipfs-api’)
+```
 
-- [circle CI](https://circleci.com)
-- [travis CI](https://travis-ci.org)
+If you use module bundler that understands ES2015 like webpack@2 or
+rollup you can use this syntax to get the original ES2015 source.
 
-## Dependencies management
+```js
+const API = require(‘ipfs-api/src’)
+```
 
-We use:
+### FAQ
 
-- [david-dm](https://david-dm.org/)
-- [greenkeeper](http://greenkeeper.io/) to keep your deps up to date.
+#### Why are you not using XYZ?
+
+There are two possibilities, either it didn’t work out for us, or
+we don’t know about it. If you think we might have missed it please tell us,
+but please believe us if we say we tried and it didn’t work for us.
+
+#### Ugh, gulp why not use simple npm scripts?
+
+Gulp is not a hard dependency, it’s just a simple way to structure our tasks
+at the moment. Usually projects only depend on the dignified binaries
+completely hiding the fact that we are using gulp under the hood. So we are
+free if we want to switch it out without any issues. We all enjoy npm
+scripts, and are using them to call the dignified binaries, but there
+is no nice way of sharing them yet.
+
+#### Where are all the semicolons?
 
-## Tasks and task runners
+Our linting rules are compatible with [standard](https://github.com/feross/standard)
+which has many examples on documentation on this. Please go there and read it.
 
-We use a mix of `npm scripts` for small tasks (small projects) and for more complex tasks, our task runner of choice is [gulp](http://gulpjs.com/).
+#### Why are you bothering with ES2015 and all this build setup?
 
-We also use [precommit](https://www.npmjs.com/package/pre-commit) to enforce running code style verification and tests on every commit.
+We want to see the web move forward, and some of us enjoy writing
+their JavaScript with things like const and arrow functions.
 
-# Project example
+#### Why are doing this?
 
-// WIP
+Because it saves us hours every single day. Hours in which we don’t
+have to think about these things or argue with someone about why we are doing it.
 
 # Contributing
 
```

Comment:
Sure, I don't have these references though I'm afraid. Webpack didn't "win" in the sense of speed. And I do not have any statistics of how many packages are shifting to standard codestyle.